client/src/pages/locations/GeoLocation.js

```diff
@@ -249,35 +249,65 @@ const LatLonFormField = ({
     </FormGroup>
   )
 
-  function setParsedLatLng(latVal, lngVal) {
-    setFieldTouched("lat", true, false)
-    setFieldTouched("lng", true, false)
-    setFieldValue("lat", parseCoordinate(latVal))
-    setFieldValue("lng", parseCoordinate(lngVal))
+  // Don't parse in onChange, it limits user
+  // displayedCoordinate is in read-only field, no need to validate
+  function setLatOnChange(val) {
+    setFieldValue("lat", val)
+    setFieldValue(
+      "displayedCoordinate",
+      convertLatLngToMGRS(val, coordinates.lng),
```

Comment:
After one of my previous remarks you can just do
```suggestion
      convertLatLngToMGRS(val, lng),
```